src/gui/torrentoptionsdialog.cpp

```diff
@@ -0,0 +1,429 @@
+}
+
+int TorrentOptionsDialog::getDownloadSpeedLimit() const
+{
+    return qMax(0, m_ui->spinDownloadLimit->value());
+}
+
+int TorrentOptionsDialog::getUploadSpeedLimit() const
+{
+    return qMax(0, m_ui->spinUploadLimit->value());
+}
+
+bool TorrentOptionsDialog::useGlobalShareLimits() const
+{
+    return m_ui->radioUseGlobalShareLimits->isChecked();
+}
+
+qreal TorrentOptionsDialog::getRatio() const
+{
+    if (m_ui->buttonGroup->checkedId() == -1) // No radio button is selected
+        return -9; // invalid number to signify different torrents ratios
+    if (useGlobalShareLimits())
+        return BitTorrent::TorrentHandle::USE_GLOBAL_RATIO;
+
+    if (m_ui->radioNoLimit->isChecked() || !m_ui->checkMaxRatio->isChecked())
+        return BitTorrent::TorrentHandle::NO_RATIO_LIMIT;
+    else
+        return m_ui->spinRatioLimit->value();
+}
+
+int TorrentOptionsDialog::getSeedingTime() const
+{
+    if (m_ui->buttonGroup->checkedId() == -1) // No radio button is selected
+        return -9; // invalid number to signify different torrents seeding times
+    if (useGlobalShareLimits())
+        return BitTorrent::TorrentHandle::USE_GLOBAL_SEEDING_TIME;
+
+    if (m_ui->radioNoLimit->isChecked() || !m_ui->checkMaxTime->isChecked())
+        return  BitTorrent::TorrentHandle::NO_SEEDING_TIME_LIMIT;
+    else
+        return m_ui->spinTimeLimit->value();
+}
+
+void TorrentOptionsDialog::handleRatioTypeChanged()
+{
+    m_ui->checkMaxRatio->setEnabled(m_ui->radioTorrentLimit->isChecked());
+    m_ui->checkMaxTime->setEnabled(m_ui->radioTorrentLimit->isChecked());
+
+    m_ui->spinRatioLimit->setEnabled(m_ui->radioTorrentLimit->isChecked() && m_ui->checkMaxRatio->isChecked());
+    m_ui->spinTimeLimit->setEnabled(m_ui->radioTorrentLimit->isChecked() && m_ui->checkMaxTime->isChecked());
+}
+
+bool TorrentOptionsDialog::isDisableDHTChecked() const
+{
+    return m_ui->checkDisableDHT->isChecked();
+}
+
+bool TorrentOptionsDialog::isDisablePEXChecked() const
+{
+    return m_ui->checkDisablePEX->isChecked();
+}
+
+bool TorrentOptionsDialog::isDisableLSDChecked() const
+{
+    return m_ui->checkDisableLSD->isChecked();
+}
+
+void TorrentOptionsDialog::handleUpSpeedLimitChanged()
+{
+    m_ui->spinUploadLimit->setMinimum(0);
+    m_ui->spinUploadLimit->setSpecialValueText(QString::fromUtf8(C_INFINITY));
+    disconnect(m_ui->spinUploadLimit, qOverload<int>(&QSpinBox::valueChanged)
+                   , this, &TorrentOptionsDialog::handleUpSpeedLimitChanged);
+}
+
+void TorrentOptionsDialog::handleDownSpeedLimitChanged()
+{
+    m_ui->spinDownloadLimit->setMinimum(0);
+    m_ui->spinDownloadLimit->setSpecialValueText(QString::fromUtf8(C_INFINITY));
+    disconnect(m_ui->spinDownloadLimit, qOverload<int>(&QSpinBox::valueChanged)
+                   , this, &TorrentOptionsDialog::handleDownSpeedLimitChanged);
+}
+
+void TorrentOptionsDialog::handleDisableDHTChanged()
+{
+    m_ui->checkDisableDHT->setTristate(false);
```

Comment:
Is it worth to change this behavior? why not use the default behavior from Qt? We can have less code then.

Reply:
I don't follow. If i don't do this, `PartiallyChecked` will be available to users when toggling the checkbox, if that's the checkbox's startup state. Or you're saying there's an easier way to do this?

Reply:
>If i don't do this, PartiallyChecked will be available to users when toggling the checkbox, if that's the checkbox's startup state. 

Yes, is there a reason to avoid this behavior? If a user misclicked a checkbox, at least he will have the chance to revert it instead of close/reopen the dialog.

Reply:
I didn't imagine someone would want this but i guess you're right. Changed it back to default.